service/history/engine/engineimpl/historyEngine.go

```diff
@@ -2495,7 +2497,9 @@ func (e *historyEngineImpl) SignalWorkflowExecution(
 			if _, err := mutableState.AddWorkflowExecutionSignaled(
 				request.GetSignalName(),
 				request.GetInput(),
-				request.GetIdentity()); err != nil {
+				request.GetIdentity(),
+				request.GetRequestID(),
```

Comment:
let's make sure new lines have test coverage

Reply:
Updated some tests to cover the new field. For those not covered, we can fix them in the fixit week. Because I don't think it's efficient for me to read the code for hours and only write unit tests for a few functions that only touched by the PR.